openedx/features/content_tagging/tests/test_rules.py

```diff
@@ -101,37 +101,88 @@ def _expected_users_have_perm(
         assert self.learner.has_perm(perm) == learner_perm
         assert self.learner.has_perm(perm, obj) == learner_obj
 
+    def _expected_users_have_perm_taxonomy(
```

Comment:
As noted above, if I'd written these tests more legibly to begin with, we wouldn't have to do this here. Will address as part of https://github.com/openedx/modular-learning/issues/89.